component/block_cache/block_cache.go

```diff
@@ -206,22 +208,23 @@ func (bc *BlockCache) Configure(_ bool) error {
 		bc.blockSize = uint64(conf.BlockSize * float64(_1MB))
 	}
 
-	bc.memSize = uint64(4192) * _1MB
+	var sysinfo syscall.Sysinfo_t
+	err = syscall.Sysinfo(&sysinfo)
+	if err != nil {
+		panic(err)
```

Comment:
We shall not panic here, instead return back error as we do in all configure methods.

Reply:
We can log the error here and assign memSize with some default value.